anesthetic/utils.py

```diff
@@ -1,11 +1,24 @@
 """Data-processing utility functions."""
 import numpy
 import pandas
+from scipy.special import logsumexp
 from scipy.interpolate import interp1d
 from scipy.stats import zscore, norm
 from matplotlib.tri import Triangulation
 
 
+def logsumexpinf(a, b, **kwargs):
+    """Handles -inf in weighted logsumexp.
+
+    If a=-inf in `log(sum(b * exp(a))` then we can set b=0, using:
+
+    limit x->-oo: x * exp(x) = 0
+
+    """
+    b = numpy.where(a == -numpy.inf, 0, b)
+    return logsumexp(a, b=b, **kwargs)
```

Comment:
Does this function make sense? I'm effectively setting the terms in the sum of logsumexp with `a=-inf` to zero.

Reply:
I'm pretty sure it does. This is similar to the idea that `0*log(0)==0`, i.e. `(anything)*exp(-inf)=0`. Ideally this function should have the same call signature as scipy.logsumexp, and I'd probably just call it logsumexp, and note that `anesthetic.utils.logsumexp` has a slightly more sensible default.

Reply:
> Ideally this function should have the same call signature as scipy.logsumexp, and I'd probably just call it logsumexp, and note that `anesthetic.utils.logsumexp` has a slightly more sensible default.

Adapted accordingly in [bfe6cfb](https://github.com/williamjameshandley/anesthetic/pull/71/commits/bfe6cfbadface8b1ed56142df2fc0a51490efde0).

Reply:
This isn't quite the same signature, since you've converted the positional `*args` to `**kwargs`, which means that if somebody tried calling `anesthetic.utils.logsumexp(a, axis)` they would get unexpected behaviour

Reply:
Corrected signature in [5ef79fa](https://github.com/williamjameshandley/anesthetic/pull/71/commits/5ef79fa69a8276d1f68fa0c8d2f39cb8fc3d654e).